Math/NumberTheory/GCD.hs

```diff
@@ -245,3 +251,31 @@ cw16 (W16# x#) (W16# y#) = coprimeWord# x# y#
 
 cw32 :: Word32 -> Word32 -> Bool
 cw32 (W32# x#) (W32# y#) = coprimeWord# x# y#
+
+-- | The input list is assumed to be a factorisation of some number
+-- into a list of powers of (possibly, composite) non-zero factors. The output
+-- list is a factorisation of the same number such that all factors
+-- are coprime. Such transformation is crucial to continue factorisation
+-- (lazily, in parallel or concurrent fashion) without
+-- having to merge multiplicities of primes, which occurs more than in one
+-- composite factor.
+--
+-- > > splitIntoCoprimes [(360, 1), (210, 1)]
```

Comment:
took me a moment to think through this, but its nicely written

Reply:
I have added one more (less complicated) example.

Reply:
ok, that def helps